source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,204 @@
+
+.. code-block:: json
+
+   { "_id": 2, "color": "green", "qty": 8 }
+   { "_id": 3, "color": "purple", "qty": 4 }
+
+After the owner runs this query, they find two orders that matched the
+criteria. 
+
+For a runnable ``find()`` example, see our :doc:`Find Multiple
+Documents </usage-examples/find>` page. 
+
+.. _retrieve-aggregate:
+
+Aggregate
+~~~~~~~~~
+
+Use the aggregate operation to perform the stages in an aggregation
+pipeline.  An aggregation pipeline is a multi-staged transformation that
+produces an aggregated result. 
+
+To perform an aggregate operation, call the ``aggregate()`` method on an
+instance of a ``MongoCollection``. This method accepts aggregation
+expressions to run in sequence. To perform aggregations, you can
+define aggregation stages that specify how to match documents, rename
+fields and group values. For more information, see our
+:doc:`Aggregation </fundamentals/aggregation>` guide.
+
+Example
+```````
+
+The owner would like to know which paint color is the most purchased
+(highest quantity sold) from their :ref:`paint_order collection <retrieve-paint-order-collection>`.
+
+To address the scenario, the owner creates an aggregation pipeline that:
+
+- Matches all the documents in the ``paint_order`` collection
+- Groups orders by colors 
+- Sums up the quantity field by color
+- Orders the results by highest-to-lowest quantity
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Retrieve.java
+   :language: java
+   :dedent:
+   :start-after: begin aggregateExample
+   :end-before: end aggregateExample
+
+The following shows the output of the aggregation specified above:
+
+.. code-block:: json
+
+   { "_id": "green", "qty": 19 }
+   { "_id": "purple", "qty": 14 }
+
+After the owner runs the aggregation, they find that "green" is the most
+purchased color.
+
+For more information on how to construct an aggregation pipeline, see
+the MongoDB server manual page on :manual:`Aggregation </aggregation>`.
+
+.. _retrieve-watch:
+
+Change Streams
+~~~~~~~~~~~~~~
+
+Open a change stream to monitor real-time changes to a collection.
+Change streams can subscribe to specific types of data changes and 
+immediately produce change events. For more information on change
+streams, see the MongoDB server manual page on :manual:`Change Streams
+</changeStreams>`. 
+
+To open a change stream, call the ``watch()`` method on an instance of a
+``MongoCollection``, ``MongoDatabase`` or ``MongoClient``. 
+You can pass an aggregation pipeline to this method to notify you as
+soon as a change occurs.
+
+The instance you call the ``watch()`` method on determines the scope of
+events which the change stream listens for. If you call ``watch()`` on a
+``MongoCollection``, the method monitors a collection. If you call
+``watch()`` on a ``MongoDatabase``, the method monitors all the
+collections in the database. If you call ``watch()`` on a
+``MongoClient``, the method monitors a standalone MongoDB instance, a
```

Comment:
Note: standalone MongoDB instances don't currently support change streams.

See: https://jira.mongodb.org/browse/SERVER-35855
and https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#single-server-topologies